docs/@docs/label-rendering/text_renderer_class_diagram.mmd

```diff
@@ -0,0 +1,65 @@
+classDiagram
+
+    class TextElement
+    class TextElementGroup
+    class TextElementState
+    class TextElementGroupState
+    class TextElementStateCache
+    class TextElementsRenderer
+    class TextCanvasRenderer
+    class TextCanvas
+    class FontCatalog
+    class PoiRenderer
+    class PoiManager
+    class ScreenCollisions
+    class TextStyleCache
+    class Theme
+    class TextStyleDefinition
+    class TextRenderStyle
+    class TextLayoutStyle
+    class PoiRenderBuffer
+
+    TextElementsRenderer --> PoiManager
+    PoiManager --> PoiInfo: create
+    TextElementsRenderer --> ScreenCollisions
+    TextElementsRenderer --> "# font catalogs" TextCanvasRenderer
+    TextCanvasRenderer --> TextCanvas
+    TextCanvasRenderer --> PoiRenderer
+    PoiRenderer --> PoiInfo : render
+    PoiRenderer --> PoiRenderBuffer
+    PoiRenderBuffer --> "# icon textures" PoiRenderBufferBatch
+    PoiRenderBufferBatch --> BoxBuffer
+    PoiInfo --> PoiRenderBufferBatch : has assigned
+    BoxBuffer --> BoxBufferMesh
+    BoxBufferMesh --> IconMaterial
+    THREE_RawShaderMaterial <|-- IconMaterial
+    BoxBufferMesh --> THREE_BufferGeometry
+    THREE_Mesh <|-- BoxBufferMesh
+    TextCanvas --> FontCatalog
+    TextCanvas --> LineTypesetter
```

Comment:
this looks very strange, I used the live editor and managed to get the LineTypeSetter & PathTypesetter to be outside of the PoiRenderer.

Reply:
See: https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiY2xhc3NEaWFncmFtXG5cbiAgICBjbGFzcyBUZXh0RWxlbWVudFxuICAgIGNsYXNzIFRleHRFbGVtZW50R3JvdXBcbiAgICBjbGFzcyBUZXh0RWxlbWVudFN0YXRlXG4gICAgY2xhc3MgVGV4dEVsZW1lbnRHcm91cFN0YXRlXG4gICAgY2xhc3MgVGV4dEVsZW1lbnRTdGF0ZUNhY2hlXG4gICAgY2xhc3MgVGV4dEVsZW1lbnRzUmVuZGVyZXJcbiAgICBjbGFzcyBUZXh0Q2FudmFzUmVuZGVyZXJcbiAgICBjbGFzcyBUZXh0Q2FudmFzXG4gICAgY2xhc3MgRm9udENhdGFsb2dcbiAgICBjbGFzcyBQb2lSZW5kZXJlclxuICAgIGNsYXNzIFBvaU1hbmFnZXJcbiAgICBjbGFzcyBTY3JlZW5Db2xsaXNpb25zXG4gICAgY2xhc3MgVGV4dFN0eWxlQ2FjaGVcbiAgICBjbGFzcyBUaGVtZVxuICAgIGNsYXNzIFRleHRTdHlsZURlZmluaXRpb25cbiAgICBjbGFzcyBUZXh0UmVuZGVyU3R5bGVcbiAgICBjbGFzcyBUZXh0TGF5b3V0U3R5bGVcbiAgICBjbGFzcyBQb2lSZW5kZXJCdWZmZXJcblxuICAgIFRleHRFbGVtZW50c1JlbmRlcmVyIC0tPiBQb2lNYW5hZ2VyXG4gICAgUG9pTWFuYWdlciAtLT4gUG9pSW5mbzogY3JlYXRlXG4gICAgVGV4dEVsZW1lbnRzUmVuZGVyZXIgLS0-IFNjcmVlbkNvbGxpc2lvbnNcbiAgICBUZXh0RWxlbWVudHNSZW5kZXJlciAtLT4gXCIjIGZvbnQgY2F0YWxvZ3NcIiBUZXh0Q2FudmFzUmVuZGVyZXJcbiAgICBUZXh0Q2FudmFzUmVuZGVyZXIgLS0-IFRleHRDYW52YXNcbiAgICBUZXh0Q2FudmFzUmVuZGVyZXIgLS0-IFBvaVJlbmRlcmVyXG4gICAgUG9pUmVuZGVyZXIgLS0-IFBvaUluZm8gOiByZW5kZXJcbiAgICBQb2lSZW5kZXJlciAtLT4gUG9pUmVuZGVyQnVmZmVyXG4gICAgUG9pUmVuZGVyQnVmZmVyIC0tPiBcIiMgaWNvbiB0ZXh0dXJlc1wiIFBvaVJlbmRlckJ1ZmZlckJhdGNoXG4gICAgUG9pUmVuZGVyQnVmZmVyQmF0Y2ggLS0-IEJveEJ1ZmZlclxuICAgIFBvaUluZm8gLS0-IFBvaVJlbmRlckJ1ZmZlckJhdGNoIDogaGFzIGFzc2lnbmVkXG4gICAgQm94QnVmZmVyIC0tPiBCb3hCdWZmZXJNZXNoXG4gICAgQm94QnVmZmVyTWVzaCAtLT4gSWNvbk1hdGVyaWFsXG4gICAgVEhSRUVfUmF3U2hhZGVyTWF0ZXJpYWwgPHwtLSBJY29uTWF0ZXJpYWxcbiAgICBCb3hCdWZmZXJNZXNoIC0tPiBUSFJFRV9CdWZmZXJHZW9tZXRyeVxuICAgIFRIUkVFX01lc2ggPHwtLSBCb3hCdWZmZXJNZXNoXG4gICAgVGV4dENhbnZhcyAtLT4gXCIqXCIgVGV4dENhbnZhc0xheWVyXG4gICAgVGV4dENhbnZhcyAtLT4gRm9udENhdGFsb2dcbiAgICBUZXh0Q2FudmFzIC0tPiBMaW5lVHlwZXNldHRlclxuICAgIFRleHRDYW52YXMgLS0-IFBhdGhUeXBlc2V0dGVyXG4gICAgVGV4dENhbnZhc0xheWVyIC0tPiBUZXh0R2VvbWV0cnlcbiAgICBUZXh0R2VvbWV0cnkgLS0-IFwiMlwiIFRIUkVFX01lc2ggOiB0ZXh0ICYgYmdcbiAgICBUSFJFRV9NZXNoIFwiMlwiIC0tPiBUSFJFRV9CdWZmZXJHZW9tZXRyeVxuICAgIFRIUkVFX01lc2ggLS0-IFwiMlwiIFNkZlRleHRNYXRlcmlhbCA6IHRleHQgJiBiZ1xuICAgIFRIUkVFX1Jhd1NoYWRlck1hdGVyaWFsIDx8LS0gU2RmVGV4dE1hdGVyaWFsXG4gICAgVGV4dENhbnZhcyAtLT4gR2x5cGhEYXRhIDogcmVuZGVyXG4gICAgRm9udENhdGFsb2cgLi4-IEdseXBoRGF0YTogY3JlYXRlXG4gICAgVGV4dEVsZW1lbnRzUmVuZGVyZXIgLS0-IFRleHRFbGVtZW50U3RhdGVDYWNoZVxuICAgIFRleHRFbGVtZW50U3RhdGVDYWNoZSAtLT4gRHVwbGljYXRlQ2FjaGVcbiAgICBEdXBsaWNhdGVDYWNoZSAgLS0-IFRleHRFbGVtZW50U3RhdGU6IGJ5IHRleHQgb3IgZmVhdHVyZUlkXG4gICAgVGV4dEVsZW1lbnRTdGF0ZUNhY2hlICotLSBUZXh0RWxlbWVudEdyb3VwU3RhdGVcbiAgICBUZXh0RWxlbWVudEdyb3VwU3RhdGUgKi0tIFRleHRFbGVtZW50U3RhdGVcbiAgICBUZXh0RWxlbWVudEdyb3VwU3RhdGUgLS0-IFRleHRFbGVtZW50R3JvdXBcbiAgICBUZXh0RWxlbWVudEdyb3VwICotLSBUZXh0RWxlbWVudFxuICAgIFRleHRFbGVtZW50IC0tPiBHbHlwaERhdGFcbiAgICBUaGVtZSAtLT4gXCIqXCIgVGV4dFN0eWxlRGVmaW5pdGlvblxuICAgIFRleHRFbGVtZW50c1JlbmRlcmVyIC0tPiBUZXh0U3R5bGVDYWNoZVxuICAgIFRleHRTdHlsZUNhY2hlIC0tPiBUZXh0UmVuZGVyU3R5bGVcbiAgICBUZXh0U3R5bGVDYWNoZSAtLT4gVGV4dExheW91dFN0eWxlXG4gICAgVGV4dEVsZW1lbnQgLS0-IFRleHRSZW5kZXJTdHlsZVxuICAgIFRleHRFbGVtZW50IC0tPiBUZXh0TGF5b3V0U3R5bGVcbiAgICBUZXh0RWxlbWVudCAtLT4gVGV4dFN0eWxlRGVmaW5pdGlvbiA6IGJ5IG5hbWVcbiIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In0sInVwZGF0ZUVkaXRvciI6ZmFsc2V9

Reply:
done. I reodered the lines and it worked somehow.